lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -513,9 +551,15 @@ ApplyInst *swift::ArraySemanticsCall::hoistOrCopy(SILInstruction *InsertBefore,
 void swift::ArraySemanticsCall::removeCall() {
   if (getSelfParameterConvention(SemanticsCall) ==
       ParameterConvention::Direct_Owned) {
-    SILBuilderWithScope Builder(SemanticsCall);
-    Builder.createReleaseValue(SemanticsCall->getLoc(), getSelf(),
-                               Builder.getDefaultAtomicity());
+    if (!SemanticsCall->getFunction()->hasOwnership()) {
+      SILBuilderWithScope Builder(SemanticsCall);
+      Builder.emitDestroyValueOperation(SemanticsCall->getLoc(), getSelf());
+    } else {
+      auto Self = getSelf();
+      assert(isa<CopyValueInst>(Self) || isa<LoadInst>(Self));
+      Self->replaceAllUsesWithUndef();
```

Comment:
Again I'm not sure how `replaceAllUsesWithUndef` works in general. Is it assuming the call takes an owned self and is in the same blocks. Even then, what about other non-consuming uses of the array that aren't being rewritten?

Reply:
Thanks for pointing it out. I fixed this and added a test.